cmd/cstorvolumeclaim/controller.go

```diff
@@ -385,3 +391,180 @@ func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
+			//delete(newConditionSet, condition.Type)
+		}
+	}
+
+	// Append remains resize conditions.
+	for _, condition := range newConditionSet {
+		resultConditions = append(resultConditions, condition)
+	}
+
+	return resultConditions
+}
+
+func (c *CVCController) markCVCResizeFinished(cvc *apis.CStorVolumeClaim) error {
+	newCVC := cvc.DeepCopy()
+
+	newCVC.Status.Conditions = c.MergeResizeConditionsOfCVC(cvc.Status.Conditions, []apis.CStorVolumeClaimCondition{})
+	_, err := c.PatchCVCStatus(cvc, newCVC)
+	if err != nil {
+		glog.Errorf("Mark CVC %q as resize finished failed: %v", cvc.Name, err)
+		return err
+	}
+
+	glog.V(4).Infof("Resize CVC %q finished", cvc.Name)
+	c.recorder.Eventf(cvc, corev1.EventTypeNormal, string(apis.CStorVolumeClaimResizeSuccess), "Resize volume succeeded")
+
+	return nil
+}
+
+// PatchCVCStatus updates CVC status using patch api
+func (c *CVCController) PatchCVCStatus(oldCVC,
+	newCVC *apis.CStorVolumeClaim,
+) (*apis.CStorVolumeClaim, error) {
+	//patchBytes, err := getPatchData(oldCVC, newCVC)
+	//if err != nil {
+	//	return nil, fmt.Errorf("can't patch status of PVC %s as generate path data failed: %v", oldCVC.Name, err)
+	//}
+	updatedClaim, updateErr := c.clientset.OpenebsV1alpha1().CStorVolumeClaims(oldCVC.Namespace).
+		Update(newCVC)
+	//Patch(oldCVC.Name, types.MergePatchType, patchBytes, "status")
+	if updateErr != nil {
+		return nil, fmt.Errorf("can't patch status of CVC %s with %v", oldCVC.Name, updateErr)
+	}
+	return updatedClaim, nil
+}
+
+func getPatchData(oldObj, newObj interface{}) ([]byte, error) {
+	oldData, err := json.Marshal(oldObj)
+	if err != nil {
+		return nil, fmt.Errorf("marshal old object failed: %v", err)
+	}
+	newData, err := json.Marshal(newObj)
+	if err != nil {
+		return nil, fmt.Errorf("mashal new object failed: %v", err)
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, oldObj)
+	if err != nil {
+		return nil, fmt.Errorf("CreateTwoWayMergePatch failed: %v", err)
+	}
+	return patchBytes, nil
+}
+
+// resizeVolume resize the volume to request size, and update CV's capacity
+func (c *CVCController) resizeVolume(
+	cvc *apis.CStorVolumeClaim,
+	cv *apis.CStorVolume) error {
+
+	newSize := cvc.Spec.Capacity[corev1.ResourceStorage]
+	err := c.UpdateCVCapacity(cv, newSize.String())
+	if err != nil {
+		glog.Errorf("Update capacity of CV %q to %s failed: %v", cv.Name, newSize.String(), err)
+		return err
+	}
+	glog.V(4).Infof("Update capacity of CV %q to %s succeeded", cv.Name, newSize.String())
+
+	return err
+
+}
+
+// UpdateCVCapacity updates CV capacity with requested size.
+func (c *CVCController) UpdateCVCapacity(cv *apis.CStorVolume, newCapacity string) error {
+	progressCondition := apis.CStorVolumeCondition{
+		Type:               apis.CStorVolumeResizing,
+		LastTransitionTime: metav1.Now(),
+		Status:             apis.ConditionInProgress,
+	}
+
+	newCV := cv.DeepCopy()
+	newCV.Status.Conditions = append(newCV.Status.Conditions, progressCondition)
+
```

Comment:
Lets not touch CV conditions as these are owned by cstorVolume controller. cstorVolume Controller should add it and remove it when required depending on the comparison between desired and actual size of CV.

Reply:
removed .. its was for the testing purpose